src/stats_state.jl

```diff
@@ -7,9 +23,24 @@ struct OptimizationStats
     hevals::Int
 end
 
-OptimizationStats(; iterations = 0, time = 0.0, fevals = 0, gevals = 0, hevals = 0) =
+function OptimizationStats(; iterations = 0, time = 0.0, fevals = 0, gevals = 0, hevals = 0)
     OptimizationStats(iterations, time, fevals, gevals, hevals)
+end
 
+"""
+$(TYPEDEF)
+
+Stores the optimization run's state at the current iteration 
+and is passed to the callback function as the first argument.
+
+## Fields
+- `iteration`: current iteration
+- `u`: current solution
+- `objective`: current objective value
+- `gradient`: current gradient
+- `hessian`: current hessian
```

Comment:
```suggestion
- `gradient`: current gradient
- `hessian`: current hessian
```

grad, hes

Make the naming match everything else, we can't just have one thing different.